hazelcast/src/main/java/com/hazelcast/client/util/ClientStateListener.java

```diff
@@ -41,27 +45,60 @@
  * will not be useful. It is the user's responsibility to instantiate the client with the same
  * ClientConfig which was used to instantiate this helper.
  */
-public class ClientStateListener
-        implements LifecycleListener {
+public class ClientStateListener implements LifecycleListener {
     private LifecycleEvent.LifecycleState currentState = STARTING;
 
     private final Lock lock = new ReentrantLock();
     private final Condition connectedCondition = lock.newCondition();
     private final Condition disconnectedCondition = lock.newCondition();
 
+    /**
+     * Registers this instance with the provided client configuration
+     * <p>
+     * This constructor is introduced to let ClientStateListener to be used via
+     * {@link com.hazelcast.client.HazelcastClient#newHazelcastFailoverClient(ClientFailoverConfig)}
+     * <p>
+     * Listeners used in the different client configs registered to single {@link ClientFailoverConfig} should be
+     * same. It can be achieved using this constructor while the other constructor
+     * {@link ClientStateListener#ClientStateListener(ClientConfig)} does not allow that usage.
+     * <p>
+     * Note that ClientStateListener should be created after all the alternative client configs are added to the
+     * client failoverConfig.
+     * Example usage:
+     * <code>
```

Comment:
Just the `<code>` does not keep the line breaks in Javadocs. I think you should use 
```
<pre>{@code
CODE SAMPLE GOES HERE
}<pre>
```
To correctly display it in Javadocs